Texassemble/texassemble.cpp

```diff
@@ -2240,6 +2282,205 @@ int __cdecl wmain(_In_ int argc, _In_z_count_(argc) wchar_t* argv[])
             }
             break;
         }
+
+    case CMD_CUBE_FROM_HC:
+    case CMD_CUBE_FROM_VC:
+    case CMD_CUBE_FROM_VC_FNZ:
+    case CMD_CUBE_FROM_HT:
+    case CMD_CUBE_FROM_HS:
+    case CMD_CUBE_FROM_VS:
+        {
+            auto src = loadedImages.cbegin();
+            auto img = (*src)->GetImage(0, 0, 0);
+            size_t twidth = 0;
+            size_t theight = 0;
+
+            switch (dwCommand)
+            {
+            case CMD_CUBE_FROM_HC:
+            case CMD_CUBE_FROM_HT:
+                twidth = width / 4;
```

Comment:
A good safety check in each of these would be:

```
if (((width % 4) != 0) || ((height % 3) != 0))
{
    wprintf(L"WARNING: cube-from-hc expects 4:3 aspect ratio\n");
}
```

3:4, 6:1, 1:6, etc.

Reply:
I'll define ratio instead of size with switch statement.
Is that OK?

```c++
size_t ratio_w = 1;
size_t ratio_h = 1;

switch (dwCommand)
{
case CMD_CUBE_FROM_HC:
case CMD_CUBE_FROM_HT:
    // 4:3 aspect ratio
    ratio_w = 4;
    ratio_h = 3;
    break;

...

}

size_t twidth = width / ratio_w;
size_t theight = height / ratio_h;

if (((width % ratio_w) != 0) || ((height % ratio_h) != 0) || (twidth != theight))
{
    wprintf(L"WARNING: %ls expects %d:%d aspect ratio\n", dwCommand, ratio_w, ratio_h);
}
```

Reply:
Looks like that should work. Just need to make sure the warning isn't emitted for other kinds of commands.

Reply:
You're right. But it's already conditioned by another switch statement.
So, I'll use my code as it is.